x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -104,25 +108,46 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	return b, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right
-// format. It publishes the event which is then forwarded to the output. In case
-// of an error set the Error field of mb.Event or simply call report.Error().
-// It calls m.fetchTableMode() or m.fetchVariableMode() depending on the response
-// format of the query.
-func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
-	db, err := sql.NewDBClient(m.Config.Driver, m.HostData().URI, m.Logger())
-	if err != nil {
-		return fmt.Errorf("could not open connection: %w", err)
+// queryDBNames returns the query to list databases present in a server
+// as per the driver name. If the given driver is not supported, queryDBNames
+// returns an empty query.
+func queryDBNames(driver string) string {
+	switch sql.SwitchDriverName(driver) {
+	// NOTE: Add support for other drivers in future as when the need arises.
+	// dbSelector function would also required to be modified in order to add
+	// support for a new driver.
+	case "mssql":
```

Comment:
I wonder if constants for driver names would be useful for readability.

Reply:
I thought about it. Except for this function, there's no place where we use these constants so I think it is not necessary at this point.